meta-ov/recipes-apps/ovmenu-ng-skripts/files/transfer-xcsoar.sh

```diff
@@ -1,45 +1,222 @@
 #!/bin/sh
+#
+# transfer-xcsoar.sh
+# System backup transfer script to and from usbstick for Openvario and XCSoar
+#
+# Created by lordfolken         2022-02-08
+# Enhanced by 7lima & Blaubart  2022-06-19
+#
+# This backup and restore script stores all XCSoar settings and relevant
+# Openvario settings like:
+#
+# -brightness of the display
+# -rotation
+# -touch screen calibration
+# -language settings
+# -dropbear settings
+# -SSH, variod and sensord status
+# 
+# backups are stored at USB stick at:
+# openvario/backup/<MAC address of eth0>/
+# So you can store backups from more than one OV on the same stick!
 
-# Transfer script for Up/Downloadng data to usbstick
-
-USB_PATH="/usb/usbstick/openvario/"
-XCSOAR_PATH="/home/root/.xcsoar"
-
-case "$(basename "$0")" in
-	'download-all.sh')
-		SRC_PATH="$XCSOAR_PATH"
-		DEST_PATH="$USB_PATH/download/xcsoar"
-		;;
-	'upload-xcsoar.sh')
-		SRC_PATH="$USB_PATH/upload/xcsoar"
-		DEST_PATH="$XCSOAR_PATH"
-		;;
-	'upload-all.sh')
-		SRC_PATH="$USB_PATH/upload"
-		DEST_PATH="$XCSOAR_PATH"
-		;;
-	*)
-		>&2 echo 'call as download-all.sh, upload-xcsoar.sh or upload-all.sh'
-		exit 1
-esac
+echo ' [==========] Starting'
+echo ' [#=========] Wait until "DONE !!" appears before you exit!'
+
+# Provident background system buffer sync to help later syncs finish quicker
+sync&
+
+# Path where the USB stick is mounted
+USB_PATH=/usb/usbstick
+
+# XCSoar settings path
+export XCSOAR_PATH=/home/root/.xcsoar
+
+# XCSoar upload path
+XCSOAR_UPLOAD_PATH=openvario/upload/xcsoar
+
+# Backup path within the USB stick
+BACKUP=openvario/backup
+
+# MAC address of the Ethernet device eth0 to do a separate backup
+MAC=`ip li|grep -A 1 eth0|tail -n 1|cut -d ' ' -f 6|sed -e s/:/-/g`
+
+# Restore Shell Function: calls rsync with unified options. 
+# Copies all files and dirs from source recursively. Parameters:
+# $1 source
+# $2 target
+# $3 comment about type of items
+restore() {
+	if 
+		# We use --checksum here due to cubieboards not having an rtc clock
+		rsync --recursive --mkpath --checksum --quiet --progress "$1" "$2"
+		test ${RSYNC_EXIT:=$?} -eq 0
+	then
+		echo " [####======] All $3 files have been restored."
+	else 
+		>&2 echo " An rsync error $RSYNC_EXIT has occurred!"
+	fi
+	# Provident system buffer sync to help later syncs finish quicker
+	sync&
+}
+
+case `basename "$0"` in
+backup-system.sh)
+	echo ' [##========] System check ...'
+	
+	# Store SSH status 
+	if   /bin/systemctl --quiet is-enabled dropbear.socket
+	then	echo enabled
+	elif /bin/systemctl --quiet is-active  dropbear.socket
+	then	echo temporary
+	else	echo disabled  
+	fi > /home/root/ssh-status
```

Comment:
Indenting

Reply:
It's a different compact indentation style deliberately used here.
Futhermore the echo commands' output is redirected by a single if ... fi output redirection.
I welcome any counterproposal to make it even clearer.